admin/export/class-export-keywords-csv.php

```diff
@@ -0,0 +1,173 @@
+<?php
+/**
+ * @package WPSEO\Admin\Export
+ */
+
+/**
+ * Class WPSEO_Export_Keywords_CSV
+ *
+ * Exports data as returned by WPSEO_Export_Keywords_Presenter to CSV.
+ */
+class WPSEO_Export_Keywords_CSV {
+
+	/**
+	 * Exports the supplied keyword query to a CSV string.
+	 *
+	 * @param array[int][string]string $data    An array of results from WPSEO_Export_Keywords_Query::get_data.
+	 * @param array[int]string         $columns An array of columns that should be presented.
+	 *
+	 * @return string A CSV string.
+	 */
+	public function export( $data, $columns ) {
+		$csv = $this->get_headers( $columns );
+
+		foreach ( $data as $result ) {
+			$csv .= $this->format( $result, $columns );
+		}
+
+		return $csv;
+	}
+
+	/**
+	 * Returns the CSV headers based on the queried columns.
+	 *
+	 * @param array[int]string $columns The columns as returned from WPSEO_Export_Keywords_Query::get_columns.
+	 *
+	 * @return string A line of CSV.
+	 */
+	protected function get_headers( $columns ) {
+		$csv = $this->format_csv_column( __( 'ID', 'wordpress-seo' ) );
+
+		if ( in_array( 'post_title', $columns, true ) ) {
+			$csv .= ',' . $this->format_csv_column( __( 'post title', 'wordpress-seo' ) );
+		}
+
+		if ( in_array( 'post_url', $columns, true ) ) {
+			$csv .= ',' . $this->format_csv_column( __( 'post url', 'wordpress-seo' ) );
+		}
+
+		if ( in_array( 'seo_score', $columns, true ) ) {
+			$csv .= ',' . $this->format_csv_column( __( 'seo score', 'wordpress-seo' ) );
+		}
+
+		if ( in_array( 'keywords', $columns, true ) ) {
+			$csv .= ',' . $this->format_csv_column( __( 'keyword', 'wordpress-seo' ) );
+		}
+
+		if ( in_array( 'keywords_score', $columns, true ) ) {
+			$csv .= ',' . $this->format_csv_column( __( 'keyword score', 'wordpress-seo' ) );
+		}
+
+		return $csv;
+	}
+
+	/**
+	 * Formats a WPSEO_Export_Keywords_Query result as a CSV line.
+	 * In case of multiple keywords it will return multiple lines.
+	 *
+	 * @param array[string]string $result  A result as returned from WPSEO_Export_Keywords_Query::get_data.
+	 * @param array[int]string    $columns The columns as returned from WPSEO_Export_Keywords_Query::get_columns.
+	 *
+	 * @return string A line of CSV, beginning with EOL.
+	 */
+	protected function format( $result, $columns ) {
+		// If our input is malformed return an empty string.
+		if ( ! is_array( $result ) || ! array_key_exists( 'ID', $result ) ) {
+			return '';
+		}
+
+		$keywords = $this->get_array_from_result( $result, 'keywords' );
+		$keywords_score = $this->get_array_from_result( $result, 'keywords_score' );
+
+		$csv = '';
+
+		// Add at least one row plus additional ones if we have more keywords.
+		do {
+			$csv .= PHP_EOL . $this->format_csv_column( $result['ID'] );
+
+			foreach ( $columns as $column ) {
```

Comment:
I'd consider extracting this `foreach` into a separate method. Feels like there's too much going on.